extensions/podman/src/extension.ts

```diff
@@ -451,10 +452,56 @@ export async function activate(extensionContext: extensionApi.ExtensionContext):
 
   // Check on initial setup
   checkDisguisedPodmanSocket(provider);
-  // update the status of the provider if the socket is changed, created or deleted
+
+  // Update the status of the provider if the socket is changed, created or deleted
   disguisedPodmanSocketWatcher = setupDisguisedPodmanSocketWatcher(provider, getSocketPath());
   extensionContext.subscriptions.push(disguisedPodmanSocketWatcher);
 
+  // Compatibility mode status bar item
+  // only available for macOS (for now).
+  if (isMac()) {
+    // Get the socketCompatibilityClass for the current OS.
+    const socketCompatibilityMode = getSocketCompatibility();
+
+    // Create a status bar item to show the status of compatibility mode as well as
+    // create a command so when you can disable / enable compatibility mode
+    const statusBarItem = extensionApi.window.createStatusBarItem();
+    statusBarItem.text = 'Docker Compatibility';
+    statusBarItem.command = 'podman.socketCompatibilityMode';
+    statusBarItem.tooltip = 'Enable or disable Docker socket compatibility mode for Podman';
+    statusBarItem.iconClass = 'fa fa-plug';
+    statusBarItem.show();
+    extensionContext.subscriptions.push(statusBarItem);
+
+    // Create a modal dialog to ask the user if they want to enable or disable compatibility mode
+    extensionApi.commands.registerCommand('podman.socketCompatibilityMode', async () => {
```

Comment:
while testing the command was not unregistered upon save.

you need to push the result of registerCommand to `extensionContext.subscriptions.push` so it's cleaned up while deactivating the extension